src/valueObjects/errors/BaseError.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * BaseError used as the foundation for all exceptions, sets the .name property to the classname.
+ */
+export default class BaseError extends Error {

```

Comment:
`BaseError` here feels like it does very little, and doesn't really seem to be worth the extra layer of indirection/hierarchy at this stage (I'm not a fan of inheritance in general, and my skepticism goes up the more deeply it's nested). I don't see how "make sure you subclass from `BaseError`" is any improvement over "make sure your errors have a `name` set".

It may be useful in the future (though I think error composition is a better approach to this), but I'm heavily in favor of delaying abstraction until it's actually _needed_, and not abstracting in advance (which will often be the wrong abstraction). With only two custom exception classes, creating a generic `BaseError` to be a foundation for _all_ exceptions in the program based on a single property shared by the two classes (literally 1-2 lines of duplicate code) seems _very_ YAGNI. I could easily see a scenario where we _don't_ want every single error to extend from a single class, in which case `BaseError` would have to be refactored to something else.